src/test/gcp/cloudscheduler.spec.ts

```diff
@@ -0,0 +1,97 @@
+import * as _ from "lodash";
+import { expect } from "chai";
+import * as sinon from "sinon";
+import * as nock from "nock";
+import * as api from "../../api";
+
+import { cloudscheduler } from "../../gcp/";
+
+const VERSION = "v1beta1";
+
+const testJob = {
+  name: "projects/test-project/locations/us-east1/jobs/test",
+  schedule: "every 5 minutes",
+  timeZone: "America/Los_Angeles",
+  httpTarget: {
+    uri: "https://afakeone.come",
+    httpMethod: "POST",
+  },
+  retryConfig: {},
+};
+describe("cloudscheduler", () => {
+  let sandbox: sinon.SinonSandbox;
+
+  describe("createOrUpdateJob", () => {
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+      nock.cleanAll();
+    });
+
+    it("should create a job if none exists", async () => {
+      /*tslint:disable*/
+      nock(api.cloudschedulerOrigin)
```

Comment:
How is it ensured that each of these nock endpoints were called? It seems just as important to the response of `createOrReplaceJob` that each of these `nock`'d endpoints were invoked.

I think you'll want to use https://github.com/nock/nock#isdone

Reply:
Added isDone to all the tests to ensure the nocked apis are called correctly